src/samples/audio/kwd_nn_detect_test.c

```diff
@@ -0,0 +1,131 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright 2021 NXP
+//
+// Author: Cristina Feies <cristina.ilie@nxp.com>
+// Author: Viorel Suman <viorel.suman@nxp.com>
+
+#include <sof/samples/audio/kwd_nn_detect_test.h>
+#include <sof/samples/audio/kwd_nn/kwd_nn_preprocess.h>
+#include <sof/samples/audio/kwd_nn/kwd_nn_process.h>
+#include <sof/samples/audio/detect_test.h>
+#include <sof/audio/component.h>
+#include <sof/lib/wait.h>
+#include <stdint.h>
+
+static __aligned(64) uint8_t preprocessed_data[KWD_NN_CONFIG_PREPROCESSED_SIZE];
+static __aligned(8) uint8_t confidences[KWD_NN_CONFIDENCES_SIZE];
+
+static int kwd_nn_detect_postprocess(uint8_t confidences[KWD_NN_CONFIDENCES_SIZE])
+{
+	int i;
+	uint8_t max_confidence = confidences[0];
+	int max_idx = 0;
+
+	for (i = 1; i < KWD_NN_CONFIDENCES_SIZE; i++) {
+		if (max_confidence < confidences[i]) {
+			max_confidence = confidences[i];
+			max_idx = i;
+		}
+	}
+	if (max_confidence < KWD_NN_MAX_CONFIDENCE_THRESHOLD)
```

Comment:
MIN_ACCEPTABLE_CONFIDENCE or something?

Reply:
True, it is a minim threshold and if we don't reach it than we consider the answer is unknown.